tutorons/css/explain.py

```diff
@@ -24,15 +29,43 @@
 def explain(selector):
     return explainer.explain(selector)
 
+def filter_non_ascii(s):
+    if ord(s) > 127:
+        return ' '
+    return s
 
-class CssSelectorExtractor(object):
+class JavascriptSelectorExtractor(object):
 
     def __init__(self):
         self.js_string_extractor = JavascriptStringExtractor()
 
     def extract(self, node):
+
         regions = self.js_string_extractor.extract(node)
         valid_regions = [r for r in regions if is_selector(r.string)]
+        
+        return valid_regions
+
+class StylesheetSelectorExtractor(object):
+
+    def extract(self, node):
+        textfield = ''.join(map(filter_non_ascii, node.text))
+        parser = tinycss.make_parser()
+        stylesheet = parser.parse_stylesheet(textfield)
+        print(stylesheet)
+
+        if len(stylesheet.rules) == 0:
+            return []
+        else:
+            valid_regions = []
+            for rule in stylesheet.rules:
+                sel = rule.selector.as_css()
+                ss_start = textfield.find(sel)
+                if rule.declarations:
+                    valid_regions.append(Region(node, ss_start, ss_start+len(sel) -1, sel))
+            if stylesheet.errors:
+                for e in stylesheet.errors:
+                    print e
```

Comment:
Remove this print statement.  If you want to log an error, you can use

    log.error("...")